.github/doctum/apiclient-services/doctum.js.twig

```diff
@@ -0,0 +1,353 @@
+var Doctum = {
+    treeJson: {{ tree|raw }},
+    {% verbatim -%}
+    /** @var boolean */
+    treeLoaded: false,
+    /** @var boolean */
+    listenersRegistered: false,
+    autoCompleteData: null,
+    /** @var boolean */
+    autoCompleteLoading: false,
+    /** @var boolean */
+    autoCompleteLoaded: false,
+    /** @var string|null */
+    rootPath: null,
+    /** @var string|null */
+    autoCompleteDataUrl: null,
+    /** @var HTMLElement|null */
+    doctumSearchAutoComplete: null,
+    /** @var HTMLElement|null */
+    doctumSearchAutoCompleteProgressBarContainer: null,
+    /** @var HTMLElement|null */
+    doctumSearchAutoCompleteProgressBar: null,
+    /** @var number */
+    doctumSearchAutoCompleteProgressBarPercent: 0,
+    /** @var autoComplete|null */
+    autoCompleteJS: null,
+    querySearchSecurityRegex: /([^0-9a-zA-Z:\\\\_\s])/gi,
+    buildTreeNode: function (treeNode, htmlNode, treeOpenLevel) {
+        var ulNode = document.createElement('ul');
+        for (var childKey in treeNode.c) {
```

Comment:
Usually in Modern JS, `for in` tends to be a touchy subject. To my understanding is because an array could contain certain any properties on the `prototype` chain and the `for in` loop picks them up. I believe now is common to use `for of` instead but it's been a while.

https://medium.com/@bennythedev/the-difference-between-for-in-and-for-of-in-javascript-c50f40438af3
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
This file is just a copy of https://github.com/code-lts/doctum/blob/main/src/Resources/themes/default/doctum.js.twig

We have to add `.ns.html` to namespace files so that they aren't overwritten by class files with the same name (yeah, seems like a pretty big oversight by this docs generator 😢 )

In order to get that to work, I had to add the logic to add `.ns.html` to the links generated in this file (see [here](https://github.com/googleapis/google-api-php-client-services/pull/5455/files#diff-3574982b2505b899268e78936ab96c5894586421a9284310fe4933683354d0dfR53))

Reply:
Thanks for the clarification!